src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -1080,6 +1128,19 @@ public int OnBeforeUnloadProject(IVsHierarchy pRealHierarchy, IVsHierarchy pStub
 
         public int OnAfterOpenSolution(object pUnkReserved, int fNewSolution)
         {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            if (ErrorHandler.Succeeded(_vsSolution.GetProperty((int)__VSPROPID.VSPROPID_SolutionDirectory, out object solutionDirectory)))
+            {
+                _solutionDirectory = (string)solutionDirectory;
+            }
+            else
+            {
+                // This may happen for "temporary" solutions - a file is open or created without a project or solution
+                _solutionDirectory = null;
+            }
```

Comment:
How about assigning `_solutionDirectory to null` above the if statement so that we can skip the else block?

Reply:
While it would reduce lines in the text file, given the non-intuitive behaviour (solution is open, but there is no solution directory), having the comment in the else block makes it much clearer, in my opinion.